lib/datadog/core/configuration.rb

```diff
@@ -81,18 +81,22 @@ def configure
         configuration = self.configuration
         yield(configuration)
 
+        start_telemetry = false
+
         safely_synchronize do |write_components|
           write_components.call(
             if components?
               replace_components!(configuration, @components)
             else
               components = build_components(configuration)
-              components.telemetry.started!
+              start_telemetry = true
               components
             end
           )
         end
 
+        components.telemetry.started! if start_telemetry
```

Comment:
Telemetry doesn't have to start inside the global. In fact, while we're factoring it, I triggered a deadlock a few times while trying to read global information for telemetry purposes.